featurebyte/api/context.py

```diff
@@ -0,0 +1,80 @@
+"""
+UseCase module
+"""
+from __future__ import annotations
+
+from typing import Any, List
+
+from featurebyte.api.savable_api_object import SavableApiObject
+from featurebyte.common.doc_util import FBAutoDoc
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.models.context import ContextModel
+from featurebyte.schema.context import ContextCreate, ContextUpdate
+
+
+class Context(SavableApiObject):
+    """
+    Context class to represent a Context in FeatureByte.
+
+    A context defines the circumstances in which features are expected to be served.
+    """
+
+    __fbautodoc__ = FBAutoDoc(proxy_class="featurebyte.Context")
+
+    # class variables
+    _route = "/context"
+    _list_schema = ContextModel
+    _get_schema = ContextModel
+    _update_schema_class = ContextUpdate
+    _list_fields = [
+        "name",
+        "entity_ids",
+        "description",
+    ]
+
+    # pydantic instance variable (public)
+    entity_ids: List[PydanticObjectId]
+
+    def _get_create_payload(self) -> dict[str, Any]:
+        """
+        Get the payload for creating a new Context.
+
+        Returns
+        -------
+        dict[str, Any]
+        """
+        data = ContextCreate(**self.dict(by_alias=True))
+        return data.json_dict()
+
+    @classmethod
+    def create(
+        cls,
+        name: str,
+        entity_ids: List[PydanticObjectId],
```

Comment:
do we want to take in entity names, instead of entity IDs? the [spec](https://featurebyte.atlassian.net/wiki/spaces/DEV/pages/16809996/Context#%5CuD83D%5CuDDD2-Requirements) seems to suggest taking in entity names instead, which also seems to be more consistent w/ the rest of the SDK (i.e. users shouldn't have to directly worry about the actual IDs most of the time)

Reply:
good point. I think it should probably using Entity objects instead. will make the change